app/models/landing_page/block/statistics.rb

```diff
@@ -40,15 +39,28 @@ def attachment
 
     def csv_rows
       @csv_rows ||= begin
-        rows = if attachment
-                 CSV.new(URI.parse(attachment.url).open, headers: true).map(&:to_h)
-               else
-                 # SCAFFOLDING
-                 csv_file_path = Rails.root.join("#{STATISTICS_DATA_PATH}/#{data['csv_file']}")
-                 CSV.read(csv_file_path, headers: true).map(&:to_h)
-               end
-        rows.each(&:deep_symbolize_keys!)
+        rows = opened_csv
+        rows.map(&:to_h).each(&:deep_symbolize_keys!)
       end
     end
+
+    def csv_headers
+      opened_csv.headers
+    end
+
+    def opened_csv
+      return csv_from_url if attachment
+
+      csv_from_file
+    end
+
```

Comment:
Is it worth memoizing opened_csv? Otherwise a call to rows potentially reads / fetches it twice (once when it sets the variable name, then again when it calls csv_rows)